### DISCUSSION TITLE
Not able to fetch more than 10 rows from Milvus collection

### DISCUSSION THREAD

**Discussion by mohit19906** [milvus-io/milvus#26856] [Q&A and General discussion]
I am using Milvus as a vector store; I am facing an issue regarding the number of rows extracted from the collection. Please find below the method that is being used,

(Library used - pymilvus)

   search_param = {

    "data": [Ques_embedd],

    "anns_field": "embeddings", 

    "param": {"metric_type": "IP", "params": {"nprobe": 10}},

    "limit": 100, #no. of chunks

     "output_fields":["document","id"],

   }

   res = coll.search(**search_param)

 

Here, we have specified limit=100, so we expect 100 rows from the collection to be extracted but every time we are only getting 10 rows extracted irrespective of the limit we specify however, if I specify

Any Limit < 10 say, 6 then 6 rows are extracted, so I have few questions for this scenario,
1. Is there any upper limit for extraction?

2. If not, how can I overcome the above problem as I need to retrieve around 100 rows ?
3. I have seen people in the community who are able to retrieve more tan 10 but are stuck at some other upper limit, so what is the solution for this problem?

Additional Info,
1. We are creating collection from dataframe method
2. Indexing is applied on the vector field

**Comment by xiaofan-luan** [comment]
> I am using Milvus as a vector store; I am facing an issue regarding the number of rows extracted from the collection. Please find below the method that is being used,
> 
> (Library used - pymilvus)
> 
> search_param = {
> 
> ```
> "data": [Ques_embedd],
> 
> "anns_field": "embeddings", 
> 
> "param": {"metric_type": "IP", "params": {"nprobe": 10}},
> 
> "limit": 100, #no. of chunks
> 
>  "output_fields":["document","id"],
> ```
> 
> }
> 
> res = coll.search(**search_param)
> 
> Here, we have specified limit=100, so we expect 100 rows from the collection to be extracted but every time we are only getting 10 rows extracted irrespective of the limit we specify however, if I specify
> 
> Any Limit < 10 say, 6 then 6 rows are extracted, so I have few questions for this scenario,
> 
> 1. Is there any upper limit for extraction?
> 2. If not, how can I overcome the above problem as I need to retrieve around 100 rows ?
> 3. I have seen people in the community who are able to retrieve more tan 10 but are stuck at some other upper limit, so what is the solution for this problem?
> 
> Additional Info,
> 
> 1. We are creating collection from dataframe method
> 2. Indexing is applied on the vector field

how many entities are there in your collection?

we have limit but the the upper limit is very large(16384)

I think it is likely to be not enough entities in your collection

**Reply by mohit19906** [reply]
@xiaofan-luan We have thousands of rows in the collection but we get only 10 rows at max as part of vector search result. Please let us know if you require other details from our end.

**Comment by yhmo** [comment]
Possible reason: the collection has lots of items with duplicated ID.
